src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -57,11 +66,13 @@ public static GivrUser getUserByIdFromDatastore(String userId) {
 
     boolean isMaintainer = false;
 
-    if (userResult.size() == 1) {
+    if (userResult.size() < 1) {
+      addNewUserToDatastore(userId, isMaintainer);
```

Comment:
This stores each new User to the Datastore. I chose to implement this here in GivrUser.java instead of in the AuthenticateServlet.

Reply:
Why do you need to store each user in the datastore?

Reply:
I have a different question: would you expect a  function that starts with Get* mutates the datastore?

Reply:
To Rob, I decided to still store users as User entities in the Datastore to track their Maintainer status, instead of creating the Maintainer entity that we also discussed in the pod meeting. The podmates and I discussed that we may need to use this implementation and add onto this in the future.

To Tay, I fixed it to doPost. That makes more sense.

Reply:
What I meant was this function is called "Get"UserById. That word get usually implies that a function should be constant relative to any backing data. It's odd that the user in the datastore is created here.

Reply:
In terms of tracking maintainer status:

If a user isn't in the datastore does that imply that the user is a maintainer or does it mean that the user is not a maintainer? Can you use the non-presence as a signal?